### PR TITLE
[Rule Tuning] 3rd Party EDR - Add Crowdstrike FDR support - 3

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Part of https://github.com/elastic/ia-trade-team/issues/242

## Summary

Adjust simple (no sequence) rules to introduce support for Crowdstrike FDR. While full logic validation wasn't possible due to the lack of a test environment, the field population for each category was verified to ensure the data was correctly structured and populated as needed.

[EDR field compatibility matrix](https://docs.google.com/spreadsheets/d/1ZaRmSXIVYLO9AGXeZge3u0W938aGxbfd6Vha52Rs1_I/edit?usp=sharing) may be of help to review, although the data provided by FDR is not consistent across event categories, and not even between event actions.

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: Tuning - Guidelines

These guidelines serve as a reminder set of considerations when tuning an existing rule.

### Documentation and Context

- [ ] Detailed description of the suggested changes.
- [ ] Provide example JSON data or screenshots.
- [ ] Provide evidence of reducing benign events mistakenly identified as threats (False Positives).
- [ ] Provide evidence of enhancing detection of true threats that were previously missed (False Negatives).
- [ ] Provide evidence of optimizing resource consumption and execution time of detection rules (Performance).
- [ ] Provide evidence of specific environment factors influencing customized rule tuning (Contextual Tuning).
- [ ] Provide evidence of improvements made by modifying sensitivity by changing alert triggering thresholds (Threshold Adjustments).
- [ ] Provide evidence of refining rules to better detect deviations from typical behavior (Behavioral Tuning).
- [ ] Provide evidence of improvements of adjusting rules based on time-based patterns (Temporal Tuning).
- [ ] Provide reasoning of adjusting priority or severity levels of alerts (Severity Tuning).
- [ ] Provide evidence of improving quality integrity of our data used by detection rules (Data Quality).
- [ ] Ensure the tuning includes necessary updates to the release documentation and versioning.

### Rule Metadata Checks

- [ ] `updated_date` matches the date of tuning PR merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive. Review to ensure the original intent of the rule is maintained.

### Testing and Validation

- [ ] Validate that the tuned rule's performance is satisfactory and does not negatively impact the stack.
- [ ] Ensure that the tuned rule has a low false positive rate.

**Review by Samirbous — APPROVED** [review]

File: rules/windows/defense_evasion_masquerading_trusted_directory.toml [hunk 1]
```diff
@@ -41,13 +42,17 @@ tags = [
...
+    "Data Source: Crowdstrike",
 ]
 timestamp_override = "event.ingested"
 type = "eql"
 
 query = '''
 process where host.os.type == "windows" and event.type == "start" and
-  process.executable : "C:\\*Program*Files*\\*.exe" and
+  process.executable : (
+    "C:\\*Program*Files*\\*.exe",
```
Comment (Aegrah) [inline rules/windows/defense_evasion_masquerading_trusted_directory.toml:53]: Any reason why there are wildcards?

File: rules/windows/defense_evasion_masquerading_trusted_directory.toml [hunk 2]
```diff
@@ -56,6 +61,18 @@ process where host.os.type == "windows" and event.type == "start" and
         "?:\\Windows\\Downloaded Program Files\\*.exe",
         "?:\\Windows\\Temp\\.opera\\????????????\\CProgram?FilesOpera*\\*.exe",
         "?:\\Windows\\Temp\\.opera\\????????????\\CProgram?Files?(x86)Opera*\\*.exe"
+  ) and
+  not (
+    event.dataset == "crowdstrike.fdr" and
```
Comment (Aegrah) [inline rules/windows/defense_evasion_masquerading_trusted_directory.toml:66]: Just curious, what is this about?

**Review by Aegrah — APPROVED** [review]
Just two questions. For the last 3 PRs, would it make sense changing the operators to be more efficient as well?

So changing `:` to `like` etc.

File: rules/windows/defense_evasion_masquerading_trusted_directory.toml (hunk 1 above)
Comment (w0rk3r) [reply rules/windows/defense_evasion_masquerading_trusted_directory.toml:53]: To match directories that masquerade as Program Files, not the original

File: rules/windows/defense_evasion_masquerading_trusted_directory.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/windows/defense_evasion_masquerading_trusted_directory.toml:66]: To not execute these wildcards unless the dataset is Crowdstrike, so we don't make the query more expensive by adding it

**Comment by w0rk3r** [comment]
> For the last 3 PRs, would it make sense changing the operators to be more efficient as well?

As we are dealing with multiple vendors and data sources, assuming they share a pattern when populating this data is very risky. We can take a look at that in the future, but for now we don't have enough data to work on that

**Pull request merged**